package/Arg.roc

```diff
@@ -0,0 +1,98 @@
+module [
+    Arg,
+    from_raw_arg,
+    to_raw_arg,
+    from_str,
+    to_str,
+    to_bytes,
+    display,
+]
+
+## An OS-aware representation of a command-line argument.
+##
+## Though we tend to think of args as Unicode strings, most operating systems
+## represent command-line arguments as lists of bytes that aren't necessarily
+## UTF-8 encoded. Windows doesn't even use bytes, but U16s.
+##
+## Most of the time, you will pass these to packages and they will handle the
+## encoding for you, but for quick-and-dirty code you can use [display] to
+## convert these to [Str] in a lossy way.
+Arg := [Unix (List U8), Windows (List U16)] implements [Eq, Inspect { toInspector: arg_inspector }]
+
+arg_inspector : Arg -> Inspector f where f implements InspectFormatter
+arg_inspector = \arg -> Inspect.str (display arg)
+
+## Wrap a raw, OS-aware numeric list into an [Arg].
+from_raw_arg : [Unix (List U8), Windows (List U16)] -> Arg
+from_raw_arg = \raw_arg -> @Arg raw_arg
+
+## Unwrap an [Arg] into a raw, OS-aware numeric list.
+##
+## This is a good way to pass [Arg]s to Roc packages.
+to_raw_arg : Arg -> [Unix (List U8), Windows (List U16)]
+to_raw_arg = \@Arg raw_arg -> raw_arg
+
+## Encode a UTF-8 [Str] to a Unix-flavored [Arg].
+from_str : Str -> Arg
+from_str = \str ->
+    @Arg (Unix (Str.toUtf8 str))
+
+## Attempt to decode an [Arg] to a UTF-8 [Str].
+to_str : Arg -> Result Str [InvalidUnicode]
+to_str = \@Arg arg ->
+    # TODO: update when Unicode -> Str conversion is ready:
+    # https://github.com/roc-lang/roc/issues/7390
+    when arg is
+        Unix unix ->
+            Str.fromUtf8 unix
+            |> Result.mapErr \_err -> InvalidUnicode
+
+        Windows _windows -> Err InvalidUnicode
+
+## Convert an [Arg] to a list of bytes.
+to_bytes : Arg -> List U8
+to_bytes = \@Arg arg ->
```

Comment:
Where is this used? 

Your pickup on my PR was about these not being bytes for Windows. Not sure if that is relevant here, just making a note in case.

Reply:
I added a new `Opt.bytes` function that converts `Arg` to bytes if someone wants the underlying bytes from an `Arg`.

Reply:
So for Windows with its `U16` codepoints instead of bytes, I convert to bytes which is a valid thing to do, it's just sidestepping the OS awareness

Reply:
Mostly people should either use `Opt.arg` for actual handling or `Opt.str` for decoding to a proper string.